### PR TITLE
Removed product.py file and changed it to engine.py.

### PR DESCRIPTION
**Pull request by sfloam**
Updated setup.cfg to run engine.py tests
Removed test_product.py to replace with test_engine.py
Added unit test with 100% coverage for enginge (test_engine.py)
Added engine.py to calculate weight without storing product in object as before

### CODE REVIEW

File: tests/test_engine.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        #valid product_1_metadata
+        self.assertEquals(p1,{"id":"1","name":"socks","category":"footwear","price":"4.50"})
+        #valid product_2_metadata
+        self.assertEquals(p2,{"id":"2","name":"shoes","category":"footwear","price":"8.50"})
+
+    def test_getWeight(self):
+        """Testing Correct Weights"""
+
+        #invalid rectype
+        self.assertEquals(self.rec_engine_2.getWeight(self.other_prod_1),"Invalid rec_type_id")
```
Comment (cfnyu) [inline tests/test_engine.py]: Create separate unit test function for Invalid rectype

File: tests/test_engine.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+        #upsell algo test          
+        self.assertEquals(self.rec_engine_1.getWeight(self.other_prod_1),2) 
+        self.assertEquals(self.rec_engine_1.getWeight(self.other_prod_2),1) 
+        self.assertEquals(self.rec_engine_1.getWeight(self.other_prod_3),1)
+        self.assertEquals(self.rec_engine_1.getWeight(self.other_prod_4),0)
+        self.assertEquals(self.rec_engine_1.getWeight(self.other_prod_5),"Invalid metadata! Confirm you entered an id, category, and price. Also, confirm that price is a numeric value!")
```
Comment (cfnyu) [inline tests/test_engine.py]: Create separate unit test function for Invalid rectype

Comment (sfloam) [reply tests/test_engine.py]: done.

File: tests/test_engine.py (hunk 1 above)
Comment (sfloam) [reply tests/test_engine.py]: done.

**Review by cfnyu — APPROVED** [review]
Everything looks good to me. All test passed when executed locally.

**Pull request merged**